### DISCUSSION TITLE
How can I start to contribute to the tailwind ?

### DISCUSSION THREAD

**Discussion by ProPhycient5** [tailwindlabs/tailwindcss#10054] [Help]
Hi everyone, I am new to this channel and community, though I have using `Tailwind CSS` over a year. I  love this library 💕 now I feel like to contribute in the development of this product by any means. So I need your guidance, what steps do I need to follow to contribute?

Although I have read github contributing docs, where I understood the steps to be followed before creating a pull request. But I am here to know the steps before working on a core feature. 

I mean :  
**Do I need to install the whole project and run on local system ?** 
or What ??

**Comment by wongjn** [comment]
Indeed you would install the whole project and run on your local system.

**Reply by ProPhycient5** [reply]
Thanks @wongjn for the response. So you mean to say first, I have to take clone of the project and install in my local system and do `npm run dev`.

**Reply by wongjn** [reply]
You would indeed take clone of the project and install in your local system, but you would not run `npm run dev`.

**Reply by ProPhycient5** [reply]
Okay @wongjn, then how can I start contributing to the project ?

**Reply by wongjn** [reply]
If you need to test any code changes you've made, you could write a test and run a test command as defined in the `package.json`.

**Reply by ProPhycient5** [reply]
Okay @wongjn, I will do the same as suggested above.

**Comment by ProPhycient5** [comment]
Hi @wongjn, I have one doubt, how can I convert the solution to any issue in Github as my commit?

**Reply by wongjn** [reply]
[Here is some documentation on how to `commit` code in `git`](https://github.com/git-guides/git-commit)

**Reply by ProPhycient5** [reply]
Thanks, @wongjn for your response. I didn't mean this question. I know the basics about how to commit the code. 
 
 Like when you create any issue in any repo in Github, that is regarded as a commit on your profile. Similarly, when you resolve or answer to any raised issue, how that can be counted on the commits on your profile ?

**Reply by wongjn** [reply]
I'd just like to clear up some misunderstanding – when one creates an issue in a repo in Github, it is regarded as an issue activity (not a commit) on one's profile.

When one resolves or answer any raised issue, I do not believe these events are counted as commits, only commits done via `git commit` (or similar) and then pushed are counted as commits.

**Reply by ProPhycient5** [reply]
Thanks, @wongjn for clearing up my misinformation. Okay, then when one resolves or answers any raised issue, is there any way to count it as profile activity?

Suppose, If I resolve 10 issues in any repo per day, won't that be considered as any profiling activity in Github?  I mean green colored box on that day.

**Reply by wongjn** [reply]
If you resolved 10 issues in any non-fork repo, it would be considered as contribution activity in your GitHub profile contribution graph. [Read more here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile#contributions-that-are-counted).

**Reply by ProPhycient5** [reply]
Thank you @wongjn for sharing the above link. This is the one I was looking for.
